app/helpers/social_image_helper.rb

```diff
@@ -0,0 +1,55 @@
+module SocialImageHelper
+  # After this date we use SocialPreview controller directly rather than passing to URL2PNG.
+  # Keeping old URLs around since they are already generated.
+  SOCIAL_PREVIEW_MIGRATION_DATETIME = DateTime.new(2019, 3, 24)
+
+  def user_social_image_url(user)
+    if use_new_social_url?(user)
+      return user_social_preview_url(user, format: :png) if user.is_a?(User)
+      return organization_social_preview_url(user, format: :png) if user.is_a?(Organization)
+    end
+
+    GeneratedImage.new(user).social_image
+  end
+
+  def article_social_image_url(article)
+    return legacy_article_social_image(article) unless use_new_social_url?(article)
+
+    if (image = article.social_image || article.main_image || article.video_thumbnail_url)
+      return cl_image_path(image,
+                           type: "fetch",
+                           width: "1000",
+                           height: "500",
+                           crop: "imagga_scale",
+                           quality: "auto",
+                           flags: "progressive",
+                           fetch_format: "auto",
+                           sign_url: true)
+    end
+
+    article_social_preview_url(article, format: :png)
+  end
+
+  def legacy_article_social_image(article)
+    cache_key = "article-social-img-#{article}-#{article.updated_at}-#{article.comments_count}"
+
+    Rails.cache.fetch(cache_key, expires_in: 1.hour) do
+      src = GeneratedImage.new(article).social_image
+      return src if src.start_with? "https://res.cloudinary.com/"
+
+      cl_image_path(src,
```

Comment:
This call seems very similar to the one on line 19. Maybe it's worth extracting it to a method to DRY it up, or just extracting the method params which seem to repeat into a constant?

Reply:
@dabrorius Thanks! I'm not confident these params won't change again in the near future, so I think for now, is fine leaving the duplication in.